tests/Operation/WatchFunctionalTest.php

```diff
@@ -1190,6 +1197,266 @@ public function testSessionFreed()
         $this->assertNull($rp->getValue($changeStream));
     }
 
+    /**
+     * Prose test: "ChangeStream will automatically resume one time on a
+     * resumable error (including not master) with the initial pipeline and
+     * options, except for the addition/update of a resumeToken."
+     */
+    public function testResumeRepeatsOriginalPipelineAndOptions()
+    {
+        if (version_compare($this->getServerVersion(), '4.0.0', '<')) {
+            $this->markTestSkipped('failCommand is not supported');
+        }
+
+        $operation = new Watch($this->manager, $this->getDatabaseName(), $this->getCollectionName(), [], $this->defaultOptions);
+
+        $aggregateCommands = [];
+
+        $this->configureFailPoint([
```

Comment:
With #657, the `\MongoDB\Tests\FunctionalTestCase::getUri` helper was changed to extract a single mongos node from a URL containing multiple nodes to avoid issues with `failCommand` being run on a different server than any subsequent commands. Spec tests that specify `useMultipleMongoses: true` recreate the client with the original connection string to properly test that behaviour. There's also #659 which re-enables change stream tests on sharded clusters which required some more changes to the way we handle change streams in tests. Since that pull request currently triggers a segfault on shutdown in the driver, I'd revisit tests on sharded clusters after merging this PR.

Furthermore, changes necessary to test sharded transactions for [PHPLIB-374](https://jira.mongodb.org/browse/PHPLIB-374) will allow us to create fail points on a specific server, which can also be used to work around the limitation of `failCommand` on sharded clusters.